### ISSUE TITLE
Monster respawn BUG 1.48 

### ISSUE THREAD

**Issue by DimasjkTV** [suriyun-production/mmorpg-kit-docs#75]
If you start to beat the monster and your character dies, monster will cease to resurrect.
Monster’s corpse does not disappear after death. Character is resurrected in another scene. It seems this only happens if the character is resurrected in another scene. I tested the fresh project 1.48, if you die from a monster and are resurrected on the same scene, then after the monster’s death, he resurrects. 
A person can hit a monster and escape to another location. After someone kills that monster, he cannot be resurrected until the next server restart. In other words, you can kill the resurrection of monsters in all locations. Something needs to be done with this!

**Comment by insthync** [comment]
I don't understand this issues, can you show me by video ?

**Comment by insthync** [comment]
Ahhh, I see, going to fix it

**Comment by insthync** [comment]
Okay, fixed, this will be included in 1.49 update

**Comment by DimasjkTV** [comment]
> Okay, fixed, this will be included in 1.49 update
Nice! Thank you!

**Comment by DimasjkTV** [comment]
> Okay, fixed, this will be included in 1.49 update

Can you write the changes to fix this error? I don’t want to set 1.49 yet due to the "Destroy delay" component removed

**Comment by insthync** [comment]
Find `foreach (BaseCharacterEntity enemy in receivedDamageRecords.Keys)` in `BaseMonsterCharacterEntity.cs` then add codes in loop to avoid empty data

FROM

```
                foreach (BaseCharacterEntity enemy in receivedDamageRecords.Keys)
                {
                    givenRewardExp = false;
```
TO
```
                foreach (BaseCharacterEntity enemy in receivedDamageRecords.Keys)
                {
                    if (enemy == null)
                        continue;

                    givenRewardExp = false;
```
